odl/trafos/wavelet.py

```diff
@@ -41,6 +41,114 @@
+        <http://www.pybytes.com/pywavelets/ref/wavelets.html>`_.
+
+    mode : `str`
+        Signal extention mode. Possible extension modes are
+
+        'zpd': zero-padding -- signal is extended by adding zero samples
+
+        'cpd': constant padding -- border values are replicated
+
+        'sym': symmetric padding -- signal extension by mirroring samples
+
+        'ppd': periodic padding -- signal is trated as a periodic one
+
+        'sp1': smooth padding -- signal is extended according to the
+        first derivatives calculated on the edges (straight line)
+
+        'per': periodization -- like periodic-padding but gives the
+        smallest possible number of decomposition coefficients.
+
+    axes : sequence of `int`, optional
+           Dimensions in which to calculate the wavelet transform.
+    """
+    if len(shape) not in (1, 2, 3):
+        raise ValueError('shape must have length 1, 2 or 3, got {}.'
+                         ''.format(len(shape)))
+
+    size_list = [shape]
+
+    if len(shape) == 1:
+        size_list = coeff_size_list(shape, 1, wbasis, mode)
+    elif len(shape) == 2:
+        x_ax = axes.count(0)
+        y_ax = axes.count(1)
+        nx = shape[0]
+        ny = shape[1]
+        if x_ax != 0:
+            for _ in range(x_ax):
+                shp_x = pywt.dwt_coeff_len(nx, filter_len=wbasis.dec_len,
+                                           mode=mode)
+                nx = shp_x
+        else:
+            shp_x = nx
+
+        if y_ax != 0:
+            for _ in range(y_ax):
+                shp_y = pywt.dwt_coeff_len(ny, filter_len=wbasis.dec_len,
+                                           mode=mode)
+                ny = shp_y
+        else:
+            shp_y = ny
+
+        size_list.append((shp_x, shp_y))
+        size_list.append(size_list[-1])
+        size_list.reverse()
+
+    else:
+        x_ax = axes.count(0)
+        y_ax = axes.count(1)
+        z_ax = axes.count(2)
+        nx = shape[0]
+        ny = shape[1]
+        nz = shape[2]
+
+        if x_ax != 0:
+            for _ in range(x_ax):
+                shp_x = pywt.dwt_coeff_len(nx, filter_len=wbasis.dec_len,
+                                           mode=mode)
+                nx = shp_x
+        else:
+            shp_x = nx
+
+        if y_ax != 0:
+            for _ in range(y_ax):
+                shp_y = pywt.dwt_coeff_len(ny, filter_len=wbasis.dec_len,
+                                           mode=mode)
+                ny = shp_y
+        else:
+            shp_y = ny
+        if z_ax != 0:
+            for _ in range(z_ax):
+                shp_z = pywt.dwt_coeff_len(nz, filter_len=wbasis.dec_len,
+                                           mode=mode)
+                nz = shp_z
+        else:
+            shp_z = nz
+
+        size_list.append((shp_x, shp_y, shp_z))
+        size_list.append(size_list[-1])
+        size_list.reverse()
+
+    return size_list
```

Comment:
Of course, I didn't see the [the gorilla in the scene](https://www.youtube.com/watch?v=vJG698U2Mvo) :-)